content/articles/mock-a-datasource-for-react-using-json-server/index.md

```diff
@@ -0,0 +1,736 @@
+- Set it up to allow API data from links.
+- Style the application.
+- Install JSON Server in the machine.
+- Set up the JSON Server as a mock database server.
+- Configure the JSON Server data source in the project.
+- Run the application.
+
+### Understanding the json-server package
+JSON server allows almost all type of back-end requests and responses like `GET`, `POST`, `PUT`, `PATCH`, and `DELETE` methods.
+
+It has routes to access data items stored in the mock database file (JSON). Few examples are `GET /posts` for fetching all the posts, `PUT /posts/1` for updating the first post, or `DELETE /posts/1` for deleting the first post.
+
+The module allows other operations to be done on the database, such as:
+
+#### Filter
+Examples - `GET/posts?title=json-server&author=riro` or `GET/comments?author.name=riro`.
+
+These end-points filters `posts` with `title` of `json-server` and `author` name of `riro` and `comments` with the `author` name of `riro` respectively.
+
+An example is a `GET` request on `http://www.albana.com/posts?title=json-server&author=riro`.
+
+#### Pagination
+Examples - `GET/posts?_page=9&_limit=23`.
+
+This request will fetch posts from page 9, and the page limit is set to 23.
+
+An example is a GET request on `http://www.albana.com//posts?_page=9&_limit=23`.
+
+#### Sorting
+e.g. `GET/posts/5/comments?_sort=votes,likes&_order=desc,asc`. This request does the sorting of the comments by using the votes and likes in an ascending and descending order, respectively. An example is a GET request on `http://www.albana.com/posts/5/comments?_sort=votes,likes&_order=desc,asc`.
+#### Slicing
+e.g. `GET/posts/4/comments?_start=20&_limit=10`. This request truncates the comments after 10 comments starting from comment number 20. An example is a GET request on `http://www.albana.com/posts/1/comments?_start=20&_end=33`.
+#### Operators
+e.g. for getting range (`_gte`, `lte`), excluding a value(`_ne`), filtering a value(`_like`). An example is a GET request on `http://www.albana.com/posts/4/comments?_ne=sad`. It excludes any comment with the value of 'sad' while `http://www.albana.com\author_like=chris` searches for authors with a name related to 'chris' (RegEx is used).
+#### Full-text search
+e.g. `GET/posts?q=tomcat`. This request searches for the value '**tomcat**' in all the records stored. An example is a GET request on `http://www.albana.com/posts?q=tomcat`.
+#### Relationships
+e.g., the inclusion of child resources(`_embed`), the inclusion of parent resources(`_expand`), to get or create nested resources. An example is `http://www.albana.com/comments/1?_expand=post` that includes parent resource by the name 'post'.
+#### Database
+e.g. `GET\db`. This GET request brings up all database items of the specified database. An example is `http://www.albana.com/customers_db`.
+#### Homepage
+e.g. `GET/`. Using the `./public` path, one can serve the application files in the public folder. An example is `http://www.albana.com/`.
```

Comment:
Have a look at the above sub-topics and make changes accordingly

Reply:
Let me take a look at the changes man

Reply:
please confirm if all is well